extension/src/paymentHandler/payment-handler.js

```diff
@@ -1,57 +1,42 @@
 const ValidatorBuilder = require('../validator/validator-builder')
-const creditCardHandler = require('../paymentHandler/creditCard/credit-card.handler')
-const paypalHandler = require('../paymentHandler/paypal/paypal.handler')
-const kcpHandler = require('../paymentHandler/kcp/kcp-payment.handler')
-const refundHandler = require('../paymentHandler/cancel-or-refund.handler')
-const fetchPaymentMethodsHandler = require('../paymentHandler/fetch-payment-methods.handler')
+const getPaymentMethodsHandler = require('../paymentHandler/get-payment-methods.handler')
+const getOriginKeysHandler = require('../paymentHandler/get-origin-keys.handler')
 
 const paymentHandlers = {
-  creditCardHandler,
-  fetchPaymentMethodsHandler,
-  paypalHandler,
-  kcpHandler,
-  refundHandler
+  getPaymentMethodsHandler,
+  getOriginKeysHandler
 }
 
+/*
+todo(ahmet)
+- refactor the validation logic, most of the logic will be removed from validator builder.
+- find a better structure for promise handler, mapping an promise array does not seem easy to understand.
+- maybe step based validations could be done in handlePayments.
+- what if handlePayment needs to return an error result rather than update actions ?
+*/
 async function handlePayment (paymentObject) {
   const validatorBuilder = ValidatorBuilder.withPayment(paymentObject)
-  const adyenValidator = validatorBuilder
-    .validateAdyen()
+  const adyenValidator = validatorBuilder.validateAdyen()
   if (adyenValidator.hasErrors())
     // if it's not adyen payment, ignore the payment
     return { success: true, data: null }
-  const merchantReferenceValidator = validatorBuilder
-    .validateMerchantReferenceField()
-  if (merchantReferenceValidator.hasErrors())
-    return {
-      success: false,
-      data: merchantReferenceValidator.buildCtpErrorResponse()
-    }
-  const paymentMethodValidator = validatorBuilder.validatePaymentMethod()
-  if (paymentMethodValidator.hasErrors())
-    return {
-      success: false,
-      data: paymentMethodValidator.buildCtpErrorResponse()
-    }
-  const handler = _getPaymentHandler(paymentObject)
-  const handlerResponse = await handler.handlePayment(paymentObject)
-  if (handlerResponse.errors)
-    return { success: false, data: handlerResponse }
+
+  const handlers = _getPaymentHandlers(paymentObject)
+  const handlerResponses = await Promise.all(
+    handlers.map(handler => handler.handlePayment(paymentObject)))
+  const handlerResponse = {
+    actions: handlerResponses.flatMap(result => result.actions)
```

Comment:
Can it happen that one or more responses will have no actions defined? If yes than flatMap will generate an `undefined` action element into the actions array.

Reply:
You're right, error handling and validations for edge cases are not well structured (which I added as a todo). Let me check what I could do for those.

Reply:
For now, I would not expect any kind of unexpected error into the execute method, for the invalid JSON cases I added it as validation before payment handler executions:  https://github.com/commercetools/commercetools-adyen-integration/pull/232/commits/29805e1c742c6ae3cfa0cee5832ee9f247a32527